### ISSUE TITLE
E402 wrongly reported after ignored lines

### ISSUE THREAD

**Issue by csala** [PyCQA/pycodestyle#939]
I have some code that looks like this:

```python
import warnings   # noqa isort:skip
warnings.filterwarnings("ignore")  # noqa isort:skip

import some
import other
import imports
```

Up to `pycodestyle==2.50`, the `noqa` was honored by `pycodestyle` ignoring the lines as if they did not exist, and therefore not complaining on the imports below.

On `2.6.0`, the `noqa` does not work anymore and I get an `E402` error for every import I have after the warnings line.

Is this the expected behavior (could not find any mention of this in the changelogs) or is this a bug?

**Comment by asottile** [comment]
fwiw, pycodestyle still works as you describe, I suspect you're running through `flake8` -- for more details on that [there's this issue](https://gitlab.com/pycqa/flake8/-/issues/638) -- basically before a noqa could be on any line and the plugin could decide independently of flake8 whether noqa should be applied.  the latter behaviour was removed as it led to plugins doing this incorrectly (notably, pycodestyle would ignore specific-noqa) and so that strict noqa can be implemented.  now when running through flake8, flake8 is the only decider of `noqa`'s behaviour (and can ensure that `noqa: CODE123` works properly and doesn't accidentally become a blanket noqa)

The actual error appears on lines 4, 5, and 6 in your example and that's where noqa can disable E402 for flake8.  alternatively, consider using `per-file-ignores` for such files.

```console
$ flake8 t.py --extend-ignore=F401
t.py:4:1: E402 module level import not at top of file
t.py:5:1: E402 module level import not at top of file
t.py:6:1: E402 module level import not at top of file
$ pycodestyle t.py
$ 
```
